### PR TITLE
Added module Checkm2: Rapid assessment of genome bin quality using machine learning

### PR DESCRIPTION
**Pull request by harper357**
- [ x] This comment contains a description of changes (with reason)
> Added support for CheckM2 https://github.com/chklovski/CheckM2

- [x] There is example tool output for tools in the <https://github.com/MultiQC/test-data> repository or attached to this PR
> https://github.com/MultiQC/test-data/pull/341
- [x] Code is tested and works locally (including with `--strict` flag)
- [ ] Everything that can be represented with a plot instead of a table is a plot
> I based this off of the module for QUAST, which outputs a table.
- [x] Report sections have a description and help text (with `self.add_section`)
- [ ] There aren't any huge tables with > 6 columns (explain reasoning if so)
> It makes a table of 13 columns, but hides 5 of them. If I need to cut more columns, I could probably remove some, but I think they are all good to have.
- [x] Each table column has a different colour scale to its neighbour, which relates to the data (e.g. if high numbers are bad, they're red)
- [x] Module does not do any significant computational work

### CODE REVIEW

File: multiqc/search_patterns.yaml [hunk 1]
```diff
@@ -215,6 +215,8 @@ cellranger/vdj_html:
   - fn: "*.html"
     contents: '"command": "Cell Ranger", "subcommand": "vdj"'
     num_lines: 20
+checkm2:
+  fn: "*quality_report.tsv"
```
[the remark this answers is not in the document]
Comment (vladsavelyev) [reply multiqc/search_patterns.yaml]: I think checking the header line would be sufficient - updated

**Comment by harper357** [comment]
Great! Thanks @vladsavelyev !

File: multiqc/modules/checkm2/checkm2.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            "Additional_Notes",
+        )
+        parsed_data = {}
+        for line in f["f"]:
+            if line.startswith("Name\t"):
+                continue
+            column_values = line.rstrip().split("\t")
+            column_values = [None if x == "None" else x for x in column_values]
+            parsed_data[column_values[0]] = {k: v for k, v in zip(column_names, column_values) if v}
+        self.checkm2_data = parsed_data
```
Comment (harper357) [inline multiqc/modules/checkm2/checkm2.py:63]: Should this line get changed to `self.checkm2_data.update(parsed_data)` to cover if there are multiple logs?

Comment (vladsavelyev) [reply multiqc/modules/checkm2/checkm2.py:63]: Would be a good idea!

**Comment by vladsavelyev** [comment]
Thank you for the contribution!

**Pull request merged**